pandas/tests/groupby/test_function.py

```diff
@@ -103,19 +99,33 @@ def test_builtins_apply(keys, f):
     result = df.groupby(keys).apply(f)
     ngroups = len(df.drop_duplicates(subset=keys))
 
-    assert_msg = f"invalid frame shape: {result.shape} (expected ({ngroups}, 3))"
-    assert result.shape == (ngroups, 3), assert_msg
+    # GH 28549
+    # grouping keys should not be included in output
+    if isinstance(keys, list):
+        result_shape = len(df.columns) - len(keys)
+    else:
+        result_shape = len(df.columns) - 1
+
+    assert_msg = "invalid frame shape: {} (expected ({}, {}))".format(
+        result.shape, ngroups, result_shape
+    )
+    assert result.shape == (ngroups, result_shape), assert_msg
 
     tm.assert_frame_equal(
         result,  # numpy's equivalent function
         df.groupby(keys).apply(getattr(np, fname)),
     )
 
     if f != sum:
-        expected = df.groupby(keys).agg(fname).reset_index()
-        expected.set_index(keys, inplace=True, drop=False)
+        # GH 28549
+        # No longer need to reset/set index here
+        expected = df.groupby(keys).agg(fname)
```

Comment:
can you try not to use groupby here and just construct the expected result.

the comment is also not very useful as a future reader won't have context.

can you break this out to a new test